scripts/aws/manage_aws_stack.sh

```diff
@@ -0,0 +1,251 @@
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameters \
+            ParameterKey=HostInstanceType,ParameterValue="${inst_type}" \
+            ParameterKey=Machinename,ParameterValue="${stack_name}" \
+            ParameterKey=AmiId,ParameterValue="${ami}" \
+            ParameterKey=EC2Type,ParameterValue="${ec2_type}" \
+            ParameterKey=PublicKeyString,ParameterValue="$(cat "${pub_key}")" \
+            ParameterKey=StackLaunchTemplate,ParameterValue="${stack_name}-launch-template"
+    
+    echo "Waiting for stack to be created"
+    aws --region "${region}" cloudformation wait stack-create-complete --stack-name "${stack_name}"
+    echo "Stack created successfully"
+
+    local instance_id
+    # shellcheck disable=SC2016
+    instance_id="$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+        --query 'Stacks[].Outputs[?OutputKey == `InstanceId`].OutputValue' --output text)"
+    echo "Waiting for stack status to be OK"
+    aws --region "${region}" ec2 wait instance-status-ok --instance-id "${instance_id}"
+    echo "Stack status OK"
+
+    local public_ip
+    # shellcheck disable=SC2016
+    public_ip=$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+        --query 'Stacks[].Outputs[?OutputKey == `PublicIp`].OutputValue' --output text)
+
+    echo "PUBLIC IP: ${public_ip}"
+}
+
+action_delete() {
+    local -r stack_name="${1}"
+    local -r region="${2}"
+
+    aws --region "${region}" cloudformation delete-stack --stack-name "${stack_name}"
+    echo "Waiting for stack to be deleted"
+    aws --region "${region}" cloudformation wait stack-delete-complete --stack-name "${stack_name}"
+    echo "Stack deleted"
+}
+
+action_describe() {
+    local -r stack_name="${1}"
+    local -r region="${2}"
+
+    aws --region "${region}" cloudformation describe-stack-events --stack-name "${stack_name}" --output json
+}
+
+action_logs() {
+    local -r stack_name="${1}"
+    local -r region="${2}"
+    local instance_id
+    # shellcheck disable=SC2016
+    instance_id="$(aws --region "${region}" cloudformation describe-stacks --stack-name "${stack_name}" \
+        --query 'Stacks[].Outputs[?OutputKey == `InstanceId`].OutputValue' --output text)"
+
+    aws ec2 get-console-output --instance-id "${instance_id}" --output text
+}
+
+action_ami() {
+    local -r os="${1}"
+    local -r arch="${2}"
+    local -r region="${3}"
+
+    echo "Listing AMIs for:"
+    echo "OS: ${os}"
+    echo "Arch: ${arch:-"any"}"
+    echo "Region: ${region}"
+    echo "Sorting latest to oldest"
+    get_amis "${os}" "${arch}" "${region}"
+}
+
+get_amis() {
+    local -r os="${1:-""}"
+    local -r arch="${2:-""}"
+    local -r region="${3}"
+    local ami_table
+    ami_table=$(aws ec2 describe-images --region "${region}" \
+        --filters "Name=name,Values=${os^^}*HVM-*Hourly2-GP3" "Name=architecture,Values=${arch}*" \
```

Comment:
One small request because bash on macos is old...
```suggestion
        --filters "Name=name,Values=$(tr '[:lower:]' '[:upper:]' <<< "${os}")*HVM-*Hourly2-GP3" "Name=architecture,Values=${arch}*" \
```

Reply:
fixed